extensions/locals.tf

```diff
@@ -12,28 +16,47 @@ locals {
     })
   }
 
-  // select environment for each service and expand config from "*"
-  services_select_env = { for k, v in var.args.services : k => merge(v, merge(lookup(v.environments, "*", {}), lookup(v.environments, var.environment, {}))) }
+  // Select environment for each service and expand config from "*"
+  extensions_with_default_and_environment_settings_merged = {
+    for extension_name, extension_config in var.args.services :
+    extension_name => merge(
+      extension_config,
+      merge(
+        lookup(extension_config.environments, "*", {}),
+        lookup(extension_config.environments, var.environment, {})
+      )
+    )
+  }
 
-  // expand plan config
-  services_expand_plan = { for k, v in local.services_select_env : k => merge(lookup(local.plans[v.type], lookup(v, "plan", "NO-PLAN"), {}), v) }
+  // If a plan is specified, expand it to the individual settings
+  extensions_with_plan_expanded = {
+    for extension_name, extension_config in local.extensions_with_default_and_environment_settings_merged :
+    extension_name => merge(
+      lookup(
+        lookup(local.plans, extension_config.type, {}),
```

Comment:
falling back to empty map here saves us having to mess about with pointless config for the extensions that don't have plans.